### ISSUE TITLE
[SOLUTION] Cannot find data on physical drive in WIN10

### ISSUE THREAD

**Issue by blackstonespecs** [felixrieseberg/windows95#144]
[Edited for formatting, and a quick reminder about free space]

TL;DR
IDGAF read through it so you don't brick your PC!!!!!

While reading through the threads I noticed that Windows 10 users are having alot of funny (as in, haha funny) problems with locating physical storage through the emulator.

I have the solution! Here it is:

YOU CANNOT READ ANY OF YOUR PHYSICAL DRIVE DATA! IT IS IMPOSSIBLE!

The reason you cannot read your drive is actually very similar to why you need this emulator in the first place -- incompatible old technology.

All Microsoft OSes before Windows XP ran on either a FAT16 or FAT32 file system. NTFS was never even introduced to Windows yet.

Which is why, in the past, if you upgraded to Windows XP, converted your file system to NTFS then tried to downgrade back to Windows 9x your computer would act as though your hard drive didn't even exist. You would have to reformat the drive and reconvert it to the original File Allocation Table.

The windows95.img file (which is the entirety of the emulator's virtual drive) is actually encoded in FAT32 format -- the virtual drive app I use - PowerISO - cannot read the .img file, but if you open it in WinRAR you'll notice it requires you to convert the files FROM FAT32 format before extracting.
Likewise if you ADD files to the .img (which is the ONLY way I've found thus far to add data to the emulator) the files are then exported in FAT32 format, because that is the only format that both Windows 10 and Windows 95 can read.

So, to add files to the virtual drive, try the following:
1) Create a small (let's say 10Gb) partition on one of your local drives, and have it formatted to a FAT32 file system. (Let's call it the "Ghost Drive") MAKE SURE YOUR PARTITION ONLY TAKES UP _**FREE SPACE**_ OR YOU WILL CORRUPT SOMETHING!!!
2) Extract the files from any images you are trying to install from, and copy them onto the Ghost Drive in their own unique folders (eg. Extract the files from the Quake install CD into a folder on the Ghost Drive named "Quake")
3) Open windows95.img in WinRAR.
4) Add your new install folders to the .img thru WinRAR and save your changes.
5) Start the emulator from scratch.

After this, your folders should be visible. Just remember, if you are using Windows XP and above you will NEED a FAT32 partition somewhere on your physical system to transfer files to the emulator without causing crashes.

Bump this for relevence and please label "Solution in Issue." It it NOT just the difference between 16-bit and 64-bit. It's also the difference between FAT16/32 and NTFS file systems.

**Comment by RokeyKokey** [comment]
Thanks for this; was immeasurably helpful.

**Comment by eidalex** [comment]
Hi there, first of all thanks for the tutorial !
However I encounter some difficulties with the method on Windows 10. 
I have formatted a USB key in FAT32 and placed the windows95.img inside it with the fodler I would want to add to the IMG file. And then... WinRar (last version) does not want to open it. ("The archive is corrupted"... well I have downloaded two different versions it normally boots up). 7Zip on the other hand can open the IMG file but it refuses to copy folder to IMG file giving "Operation not implemented"...
Would you please help me ? As this ever worked for anyone lately ?
Thanks in advance

**Comment by blackstonespecs** [comment]
Are you able to open the archive through winrar without attempting to add
the file to the archive?

If so so long as you have the "Ghost Drive" partition set up, and it is in
FAT32 Format try this:

1)format and add the file to the ghost drive
2)while viewing the windows95.img, copy its entire content to the ghost
drive then delete all of the .img contents but KEEP the archive open.
3)add the entire contents of your ghost drive into the windows95.img and
save it. It should then display a caution window talking about converting
the new contents into FAT32. Allow the conversion and exit.

It's worked for me, but I should warn you: you must have a duplicate copy
of the original windows95.img backed up somewhere. I found that, the more
times you recompile the package this way, the more unstable the emulator
becomes over time.

**Comment by eidalex** [comment]
If as the file we are talking of the windows95.img stored in Users\....\Local\windows95\app-2.2.1\ressources\app\images\ then the answer is no. Either on a FAT32 or NTFS partition I cannot manage to open it with WinRar. The message I have (translated from French) : "the archive is damaged or corrupted". Without speaking of adding anythong to it. For me, opening an .IMG file with WinRar does not work...
Thanks for your prompt and detailed reply by the way :)

**Comment by kengranderson** [comment]
I have been able to edit the img in WinImage and OSFMount, but could never see the files in Windows95 until I did this, which I found in another issue thread:

"You may need to "Boot from scratch" after starting windows95 again."

**Comment by felixrieseberg** [comment]
Thanks for writing this up! Happily, the whole mount-the-image workflow is obsolete now. Since v5.0, getting files into the machine is a lot easier — no more mounting the disk image by hand:

- **Shared folder**: any folder on your computer shows up inside Windows 95 as drive `Z:` (Settings → Shared Folder; defaults to your Downloads folder). Download something on your real machine, open My Computer → `Z:` inside Windows, copy it to `C:\`, done.
- **CD-ROM**: mount any `.iso` and it appears as `D:`.
- **Floppy**: mount any `.img` as `A:`.

CD-ROM and floppy images can be swapped from the Machine menu while Windows is running.
